docs/proposals/initial-resources.md

```diff
@@ -0,0 +1,57 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest 1.0.x release of this document can be found
+[here](http://releases.k8s.io/release-1.0/docs/proposals/initial-resources.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+Since we want to make Kubernetes as simple as possible for its users we don’t want to require setting [Resources](../../pkg/api/v1/types.go#L696) for container by its owner. On the other hand having Resources filled is critical for scheduling decisions. Current solution to set up Resources to hardcoded value has obvious drawbacks. We need to implement a component which will set initial Resources to a reasonable value.
+
+### Design
+
+The component will be implemented as an [admission plugin](../../plugin/pkg/admission/) and invoked right before [LimitRanger](../../cluster/gce/config-default.sh#L91). For every container without Resources specified it will try to predict amount of resources that should be sufficient for it. So that a pod without specified resources will be treated as Burstable (see more details [#11713](https://github.com/GoogleCloudPlatform/kubernetes/pull/11713)). The component will set only [Request](https://github.com/GoogleCloudPlatform/kubernetes/blob/3d2d99c6fd920386eea4ec050164839ec6db38f0/pkg/api/types.go#L691) field in the first version to avoid killing containers due to OOM (however the container still may be preempted if exceeds requested resources). To make the component work with LimitRanger the estimated value will be capped by min and max possible values if defined. It will prevent from situation when the pod is rejected due to too low or too high estimation. The container won’t be marked as managed by this component in any way, however appropriate event will be exported. The predicting algorithm should be simple to not increase significantly e2e pod startup latency [#3954](https://github.com/GoogleCloudPlatform/kubernetes/pull/3954).
```

Comment:
IIUC the reasoning is:
* auto-scaler sets request
* therefore it doesn't really make sense for user to set limit
* therefore limit is un-set, which by definition means the container is burstable

You might want to make this clearer.